botocore/endpoint.py

```diff
@@ -234,41 +248,39 @@ def create_endpoint(self, service_model, region_name=None, is_secure=True,
                         response_parser_factory=None, timeout=DEFAULT_TIMEOUT):
```

Comment:
Might be better to address later, but it would be really cool if we could get rid of any arguments such as ``service_model``, ``region_name``, ``is_secure`` as a result of moving endpoint resolving logic out of the endpoint.